baseapp/abci_utils.go

```diff
@@ -0,0 +1,243 @@
+package baseapp
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/cockroachdb/errors"
+	abci "github.com/cometbft/cometbft/abci/types"
+	cmtcrypto "github.com/cometbft/cometbft/crypto"
+	cryptoenc "github.com/cometbft/cometbft/crypto/encoding"
+	cmtprotocrypto "github.com/cometbft/cometbft/proto/tendermint/crypto"
+	cmtproto "github.com/cometbft/cometbft/proto/tendermint/types"
+
+	protoio "github.com/cosmos/gogoproto/io"
+	"github.com/cosmos/gogoproto/proto"
+
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/mempool"
+)
+
+type (
+	// Validator defines the interface contract require for verifying vote extension
+	// signatures. Typically, this will be implemented by the x/staking module,
+	// which has knowledge of the CometBFT public key.
+	Validator interface {
```

Comment:
Where do these get set?

Reply:
The vote extensions handler is defined by the app, e.g. in the `app` package. It's the exact same way an app defines the prepare/process proposal handlers.

In this vote extension handler, an app would use `ValidateVoteExtensions` if it wants to. If it does want to use it, it would pass the `x/staking` keeper as an argument which fulfills the `ValidatorStore` interface.

TLDR -- this is all outside of baseapp.

Reply:
could you update the apps wit comments on this please

Reply:
update the apps?